tcping.go

```diff
@@ -391,7 +395,7 @@ func tcping(tcpStats *stats) {
 	IPAndPort := net.JoinHostPort(tcpStats.ip, tcpStats.port)
 
 	connStart := getSystemTime()
-	conn, err := net.DialTimeout("tcp", IPAndPort, oneSecond)
+	conn, err := net.DialTimeout("tcp", IPAndPort, tcpStats.timeout)
```

Comment:
There is more to this. We handle the time calculation statically, see lines 419-422 for reference (there is more).

Custom timeout values should also take those calculations in account to prevent wrong reports. The best way to see if the changes are correctly changing the overall reports is to manually cause some failures and then see if the reported timings are matching the reality.